ext_mongodb.php

```diff
@@ -853,21 +1100,51 @@ public function __debugInfo() : array
 
 final class UTCDateTime implements Type, \Serializable
 {
-	use DenySerialization;
-
 	private int $milliseconds;
 
+	static private function checkArray( array $state )
+	{
+		if (
+			!array_key_exists( 'milliseconds', $state ) ||
+			! ( is_int( $state['milliseconds'] ) || is_string( $state['milliseconds'] ) )
+		) {
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "MongoDB\BSON\UTCDateTime initialization requires \"milliseconds\" integer or numeric string field" );
+		}
+	}
+
+	public function serialize() : string
+	{
+		return serialize( [
+			'milliseconds' => (string) $this->milliseconds,
+		] );
+	}
+
+	public function unserialize(mixed $serialized) : void
+	{
+		$unserialized = unserialize( $serialized );
+		self::checkArray( $unserialized );
+		$this->__construct( $unserialized['milliseconds'] );
+	}
+
 	public function __construct(mixed $milliseconds = NULL)
 	{
 		if ($milliseconds === NULL) {
 			$this->milliseconds = floor( microtime( true ) * 1000 );
-		} elseif (is_object($milliseconds) && $milliseconds instanceof \DateTimeInterface) {
+		} elseif (is_object( $milliseconds) && $milliseconds instanceof \DateTimeInterface) {
 			$this->milliseconds = floor( (string) $milliseconds->format('U.u') * 1000 );
-		} else {
+		} elseif ( is_numeric( $milliseconds ) ) {
```

Comment:
I did notice that in your code, but couldn't really see a good (read, no really slow) way of handling this. I don't have access to atoi or strtoll in PHP land.